src/components/TweetList.js

```diff
@@ -11,7 +11,31 @@ const defaultProps = {
 };
 
 function renderTweets(tweets) {
-  return tweets.map(tweet => <Tweet tweet={tweet} key={tweet.id} />);
+
+  tweets.forEach(tweet => {
+    let imageSmall;
+
+    if(tweet['entities']['media'] !== undefined){
+      console.log(tweet['entities']);
+      imageSmall = tweet['entities']['media'][0]['media_url'] + ':small';
+      Object.assign(tweet, {imageSmall: imageSmall});
+    }
+
+    let link;
+
+    if(/http\S*/.test(tweet['text'])){
+      link = tweet['text'].match(/http\S*/g).join('')
+    }
+
+    Object.assign(tweet, {link: link})
+    tweet['text'] = tweet['text'].replace(/http\S*/g,'')
```

Comment:
Also this line is actually modifying the tweet object. Since the tweet data is coming from the redux store you should always avoid modifying any data. See these images for example of the tweet data being modified.

Store after a fetch.
![screen shot 2016-09-23 at 19 01 35](https://cloud.githubusercontent.com/assets/196567/18796225/316713c0-81c0-11e6-8f41-115a5482c5e8.png)

Store after another fetch, tweet text has been modified.
![screen shot 2016-09-23 at 19 01 42](https://cloud.githubusercontent.com/assets/196567/18796227/341cc29a-81c0-11e6-89d1-8fd95773d120.png)

The redux store data should never be modified. In this case it's being changed because you are modifying the tweet object. This would be avoided if you were doing this in the Tweet component itself, where you could create a reference to the text string itself and render what you need without modifying the tweet object.